Generated code has compilation errors

## Issue Description

Attempting to generate code based on minimal schema results in compilation error in generated code.

## Steps to Reproduce
Minimal schema:

```
type Query {
    product(
        companyId: ID!
    ): Product!
}

interface Product {
    linkCode(
        pid: ID!
    ): LinkCode
}

type LinkCode {
    html: String
}
```

## Actual Result
mvn clean compile results in:

```
[INFO] -------------------------------------------------------------
[ERROR] COMPILATION ERROR :
[INFO] -------------------------------------------------------------
[ERROR] /Users/marnold/gits/product-search-api-java-client-graphql-java-codegen/target/generated-sources/graphql1/com/cj/product_api/graphql/model/ProductResponseProjection.java:[42,47] cannot find symbol
  symbol:   class ProductLinkCodeParametrizedInput
  location: class com.cj.product_api.graphql.model.ProductResponseProjection
[ERROR] /Users/marnold/gits/product-search-api-java-client-graphql-java-codegen/target/generated-sources/graphql1/com/cj/product_api/graphql/model/ProductResponseProjection.java:[46,61] cannot find symbol
  symbol:   class ProductLinkCodeParametrizedInput
  location: class com.cj.product_api.graphql.model.ProductResponseProjection
```

## Your Environment and Setup
pom follows:
```xml
<?xml version="1.0" encoding="UTF-8"?>
<project xmlns="http://maven.apache.org/POM/4.0.0"
         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
    <modelVersion>4.0.0</modelVersion>

    <groupId>org.example</groupId>
    <artifactId>product-search-api-java-client-graphql-java-codegen</artifactId>
    <version>1.0-SNAPSHOT</version>

    <properties>
        <codegen.version>3.1.0</codegen.version>
    </properties>

    <build>
        <plugins>
            <!-- GraphQL Codegen Maven plugin -->
            <plugin>
                <groupId>io.github.kobylynskyi</groupId>
                <artifactId>graphql-codegen-maven-plugin</artifactId>
                <!-- use the latest available version:
                https://search.maven.org/artifact/io.github.kobylynskyi/graphql-codegen-maven-plugin -->
                <version>${codegen.version}</version>
                <executions>
                    <execution>
                        <id>graphqlCodegenService1</id>
                        <goals>
                            <goal>generate</goal>
                        </goals>
                        <configuration>
                            <graphqlSchemaPaths>${project.basedir}/src/main/resources/schema1.graphqls</graphqlSchemaPaths>
                            <outputDir>${project.build.directory}/generated-sources/graphql1</outputDir>
                            <modelPackageName>com.cj.product_api.graphql.model</modelPackageName>
                            <customTypesMapping>
                                <DateTime>java.util.Date</DateTime>
                            </customTypesMapping>
                            <generateApis>false</generateApis>
                            <generateClient>true</generateClient>
                        </configuration>
                    </execution>
                </executions>
            </plugin>
            <!-- GraphQL Codegen Maven plugin -->

            <plugin>
                <groupId>org.apache.maven.plugins</groupId>
                <artifactId>maven-compiler-plugin</artifactId>
                <version>3.8.1</version>
                <configuration>
                    <encoding>UTF-8</encoding>
                    <source>1.8</source>
                    <target>1.8</target>
                    <showDeprecation>true</showDeprecation>
                    <annotationProcessorPaths>
                        <annotationProcessorPath>
                            <groupId>org.projectlombok</groupId>
                            <artifactId>lombok</artifactId>
                            <version>1.18.12</version>
                        </annotationProcessorPath>
                        <annotationProcessorPath>
                            <groupId>org.mapstruct</groupId>
                            <artifactId>mapstruct-processor</artifactId>
                            <version>1.3.1.Final</version>
                        </annotationProcessorPath>
                    </annotationProcessorPaths>
                </configuration>
            </plugin>
        </plugins>
    </build>

    <dependencies>

        <dependency>
            <groupId>com.graphql-java-kickstart</groupId>
            <artifactId>graphql-spring-boot-starter</artifactId>
            <version>7.1.0</version>
        </dependency>
        <dependency>
            <groupId>com.graphql-java-kickstart</groupId>
            <artifactId>graphiql-spring-boot-starter</artifactId>
            <version>7.1.0</version>
        </dependency>
        <dependency>
            <groupId>com.graphql-java</groupId>
            <artifactId>graphql-java-extended-scalars</artifactId>
            <version>1.0.1</version>
        </dependency>

        <dependency>
            <groupId>io.github.kobylynskyi</groupId>
            <artifactId>graphql-java-codegen</artifactId>
            <!-- use the latest available version:
            https://search.maven.org/artifact/io.github.kobylynskyi/graphql-java-codegen -->
            <version>${codegen.version}</version>
        </dependency>
    </dependencies>


</project>
```

--- a/src/main/java/com/kobylynskyi/graphql/codegen/GraphQLCodegen.java
+++ b/src/main/java/com/kobylynskyi/graphql/codegen/GraphQLCodegen.java
@@ -279,8 +279,7 @@ public class GraphQLCodegen {
         for (ExtendedInterfaceTypeDefinition definition : document.getInterfaceDefinitions()) {
             generateFieldResolver(context, definition.getFieldDefinitions(), definition).ifPresent(generatedFiles::add);
         }
-        System.out.println(String.format("Generated %d definition classes in folder %s",
-                generatedFiles.size(), outputDir.getAbsolutePath()));
+        System.out.printf("Generated %d definition classes in folder %s%n", generatedFiles.size(), outputDir.getAbsolutePath());
         return generatedFiles;
     }
 
@@ -304,6 +303,13 @@ public class GraphQLCodegen {
         if (Boolean.TRUE.equals(mappingConfig.getGenerateClient())) {
             Map<String, Object> responseProjDataModel = RequestResponseDefinitionToDataModelMapper.mapResponseProjection(mappingContext, definition);
             generatedFiles.add(GraphQLCodegenFileCreator.generateFile(FreeMarkerTemplatesRegistry.responseProjectionTemplate, responseProjDataModel, outputDir));
+
+            for (ExtendedFieldDefinition fieldDefinition : definition.getFieldDefinitions()) {
+                if (!Utils.isEmpty(fieldDefinition.getInputValueDefinitions())) {
+                    Map<String, Object> fieldProjDataModel = RequestResponseDefinitionToDataModelMapper.mapParametrizedInput(mappingContext, fieldDefinition, definition);
+                    generatedFiles.add(GraphQLCodegenFileCreator.generateFile(FreeMarkerTemplatesRegistry.parametrizedInputTemplate, fieldProjDataModel, outputDir));
+                }
+            }
         }
         return generatedFiles;
     }
@@ -398,12 +404,14 @@ public class GraphQLCodegen {
     private Optional<File> generateFieldResolver(MappingContext mappingContext,
                                                  List<ExtendedFieldDefinition> fieldDefinitions,
                                                  ExtendedDefinition<?, ?> parentDefinition) {
-        List<ExtendedFieldDefinition> fieldDefsWithResolvers = fieldDefinitions.stream()
-                .filter(fieldDef -> FieldDefinitionToParameterMapper.generateResolversForField(mappingContext, fieldDef, parentDefinition))
-                .collect(toList());
-        if (!fieldDefsWithResolvers.isEmpty()) {
-            Map<String, Object> dataModel = FieldDefinitionsToResolverDataModelMapper.mapToTypeResolver(mappingContext, fieldDefsWithResolvers, parentDefinition.getName());
-            return Optional.of(GraphQLCodegenFileCreator.generateFile(FreeMarkerTemplatesRegistry.operationsTemplate, dataModel, outputDir));
+        if (Boolean.TRUE.equals(mappingConfig.getGenerateApis())) {
+            List<ExtendedFieldDefinition> fieldDefsWithResolvers = fieldDefinitions.stream()
+                    .filter(fieldDef -> FieldDefinitionToParameterMapper.generateResolversForField(mappingContext, fieldDef, parentDefinition))
+                    .collect(toList());
+            if (!fieldDefsWithResolvers.isEmpty()) {
+                Map<String, Object> dataModel = FieldDefinitionsToResolverDataModelMapper.mapToTypeResolver(mappingContext, fieldDefsWithResolvers, parentDefinition.getName());
+                return Optional.of(GraphQLCodegenFileCreator.generateFile(FreeMarkerTemplatesRegistry.operationsTemplate, dataModel, outputDir));
+            }
         }
         return Optional.empty();
     }

--- a/src/main/java/com/kobylynskyi/graphql/codegen/mapper/RequestResponseDefinitionToDataModelMapper.java
+++ b/src/main/java/com/kobylynskyi/graphql/codegen/mapper/RequestResponseDefinitionToDataModelMapper.java
@@ -64,7 +64,7 @@ public class RequestResponseDefinitionToDataModelMapper {
      */
     public static Map<String, Object> mapParametrizedInput(MappingContext mappingContext,
                                                            ExtendedFieldDefinition fieldDefinition,
-                                                           ExtendedObjectTypeDefinition parentTypeDefinition) {
+                                                           ExtendedDefinition<?, ?> parentTypeDefinition) {
         String className = MapperUtils.getParametrizedInputClassName(mappingContext, fieldDefinition, parentTypeDefinition);
         Map<String, Object> dataModel = new HashMap<>();
         // ParametrizedInput classes are sharing the package with the model classes, so no imports are needed

--- a/src/test/java/com/kobylynskyi/graphql/codegen/GraphQLCodegenFieldsResolversTest.java
+++ b/src/test/java/com/kobylynskyi/graphql/codegen/GraphQLCodegenFieldsResolversTest.java
@@ -51,6 +51,19 @@ class GraphQLCodegenFieldsResolversTest {
     }
 
     @Test
+    void generate_ParametrizedFieldsInInterface() throws Exception {
+        mappingConfig.setGenerateClient(true);
+        mappingConfig.setGenerateApis(false);
+
+        new GraphQLCodegen(singletonList("src/test/resources/schemas/parametrized-input-client.graphqls"),
+                outputBuildDir, mappingConfig, TestUtils.getStaticGeneratedInfo()).generate();
+
+        File[] files = Objects.requireNonNull(outputJavaClassesDir.listFiles());
+        assertSameTrimmedContent(new File("src/test/resources/expected-classes/request/ProductLinkCodeParametrizedInput.java.txt"),
+                getFileByName(files, "ProductLinkCodeParametrizedInput.java"));
+    }
+
+    @Test
     void generate_CustomFieldsResolvers() throws Exception {
         mappingConfig.setModelNamePrefix("Github");
         mappingConfig.setModelNameSuffix("TO");

--- a/src/test/java/com/kobylynskyi/graphql/codegen/GraphQLCodegenModelsForRootTypesTest.java
+++ b/src/test/java/com/kobylynskyi/graphql/codegen/GraphQLCodegenModelsForRootTypesTest.java
@@ -70,8 +70,8 @@ class GraphQLCodegenModelsForRootTypesTest {
 
         File[] files = Objects.requireNonNull(outputJavaClassesDir.listFiles());
         List<String> generatedFileNames = Arrays.stream(files).map(File::getName).sorted().collect(toList());
-        assertEquals(Arrays.asList("Event.java", "EventProperty.java", "EventPropertyResolver.java", "EventStatus.java",
-                "Mutation.java", "MutationResolver.java", "Query.java", "QueryResolver.java", "Subscription.java"),
+        assertEquals(Arrays.asList("Event.java", "EventProperty.java", "EventStatus.java",
+                "Mutation.java", "Query.java", "Subscription.java"),
                 generatedFileNames);
     }
 

--- a/src/test/resources/expected-classes/request/ProductLinkCodeParametrizedInput.java.txt
+++ b/src/test/resources/expected-classes/request/ProductLinkCodeParametrizedInput.java.txt
@@ -1,0 +1,41 @@
+package com.github.graphql;
+
+import com.kobylynskyi.graphql.codegen.model.graphql.GraphQLParametrizedInput;
+import com.kobylynskyi.graphql.codegen.model.graphql.GraphQLRequestSerializer;
+import java.util.StringJoiner;
+
+/**
+ * Parametrized input for field linkCode in type Product
+ */
+@javax.annotation.Generated(
+    value = "com.kobylynskyi.graphql.codegen.GraphQLCodegen",
+    date = "2020-12-31T23:59:59-0500"
+)
+public class ProductLinkCodeParametrizedInput implements GraphQLParametrizedInput {
+
+    @javax.validation.constraints.NotNull
+    private String pid;
+
+    public ProductLinkCodeParametrizedInput() {
+    }
+
+    public ProductLinkCodeParametrizedInput(String pid) {
+        this.pid = pid;
+    }
+
+    public ProductLinkCodeParametrizedInput pid(String pid) {
+        this.pid = pid;
+        return this;
+    }
+
+
+    @Override
+    public String toString() {
+        StringJoiner joiner = new StringJoiner(", ", "(", ")");
+        if (pid != null) {
+            joiner.add("pid: " + GraphQLRequestSerializer.getEntry(pid));
+        }
+        return joiner.toString();
+    }
+
+}

--- a/src/test/resources/schemas/parametrized-input-client.graphqls
+++ b/src/test/resources/schemas/parametrized-input-client.graphqls
@@ -1,0 +1,15 @@
+type Query {
+    product(
+        companyId: ID!
+    ): Product!
+}
+
+interface Product {
+    linkCode(
+        pid: ID!
+    ): LinkCode
+}
+
+type LinkCode {
+    html: String
+}

